File: Problems & Solutions/144. Binary Tree Preorder Traversal/09 January 2022/Approach 1 - Recursion/Binary Tree Preorder Traversal.py
class Solution:
    def preorderTraversal(self, root: TreeNode) -> List[int]:
        answer = []
        
        def dfs(node):
            if not node:
                return
            answer.append(node.val)
            dfs(node.left)
            dfs(node.right)
            
        dfs(root)
        return answer

File: Problems & Solutions/144. Binary Tree Preorder Traversal/09 January 2022/Approach 1 - Recursion/Binary Tree Preorder Traversal.java
class Solution {
    private List<Integer> answer = new ArrayList<>();

    private void dfs(TreeNode node) {
        if (node == null) {
            return;
        }
        answer.add(node.val);
        dfs(node.left);
        dfs(node.right);
    }
    
    public List<Integer> preorderTraversal(TreeNode root) {
        dfs(root);
        return answer;
    }
}